lib/jekyll/commands/serve.rb

```diff
@@ -51,6 +107,70 @@ def process(opts)
           start_up_webrick(opts, destination)
         end
 
+        def shutdown
+          @server.shutdown if @running
+        end
+
+        private
+        def validate_options(opts)
+          if opts["livereload"]
+            if opts["detach"]
+              Jekyll.logger.abort_with "Error:",
+                "--detach and --livereload are mutually exclusive"
+            end
+            if opts["ssl_cert"] || opts["ssl_key"]
+              # This is not technically true.  LiveReload works fine over SSL, but
+              # EventMachine's SSL support in Windows requires building the gem's
+              # native extensions against OpenSSL and that proved to be a process
+              # so tedious that expecting users to do it is a non-starter.
+              Jekyll.logger.abort_with "Error:", "LiveReload does not support SSL"
+            end
+            unless opts["watch"]
+              Jekyll.logger.warn "Using --livereload without --watch defeats the purpose"\
```

Comment:
I think it makes sense for `--livereload` to also implicitly add `--watch` to the command line options and both do the watching and the livereloading. It feels like not being pedantic about the options here is a better experience for the user.